libvast/src/system/type_registry.cpp

```diff
@@ -112,9 +112,8 @@ caf::error type_registry_state::load_from_disk() {
     auto buffer = io::read(fname);
     if (!buffer)
       return buffer.error();
-    caf::binary_deserializer source{self->system(), *buffer};
-    if (auto error = source(data))
-      return error;
+    if (!detail::legacy_deserialize(buffer.cvalue(), data))
+      return caf::make_error(ec::parse_error, "data deserialization failed");
```

Comment:
If this errors the users get to see this message, so I suggest rewording like this:

```suggestion
      return caf::make_error(ec::parse_error, "failed to load type-registry state");
```